realm/realm-library/src/objectServer/java/io/realm/SyncManager.java

```diff
@@ -285,6 +286,7 @@ private static synchronized void notifyNetworkIsBack() {
      * can leak since we don't have control over the session lifecycle.
      */
     @SuppressWarnings("unused")
+    @KeepMember
```

Comment:
`@KeepMember` needs to be on the class as well. but, wait, the class has been already annotated with `@Keep` so this is not necessary ....